src/probnum/randprocs/kernels/_kernel.py

```diff
@@ -343,138 +312,97 @@ def _evaluate(
-            :class:`Kernel` instance.
+            See "Returns" section in the docstring of :meth:`__call__`.
         """
 
-    def _kernel_broadcast_shapes(
+    def _check_shapes(
         self,
-        x0: np.ndarray,
-        x1: Optional[np.ndarray] = None,
+        x0_shape: ShapeType,
+        x1_shape: Optional[ShapeType] = None,
     ) -> ShapeType:
-        """Applies the "kernel broadcasting" rules to the input shapes.
+        """Checks input argument shapes and computes the broadcast batch shape of both
+        inputs.
 
-        A description of the "kernel broadcasting" rules can be found in the docstring
-        of :meth:`Kernel.__call__`.
-        Appropriate exceptions are raised if the shapes can not be "kernel broadcast" to
-        a common shape.
+        This function checks the shapes of the inputs to the :meth:`__call__` method and
+        it computes the `bcast_batch_shape` mentioned in the docstring.
 
         Parameters
         ----------
-        x0 :
-            First input to the covariance function.
-        x1 :
-            Second input to the covariance function.
+        x0_shape :
+            Shape of the first input to the covariance function.
+        x1_shape :
+            Shape of the (optional) second input to the covariance function.
 
         Returns
         -------
-        broadcast_input_shape : tuple of int
-            Shape of the inputs after applying "kernel broadcasting".
+        broadcast_batch_shape :
+            The `batch_shape` after broadcasting the inputs to a common shape.
 
         Raises
         -------
         ValueError
-            If the inputs can not be "kernel broadcast" to a common shape.
+            If one of the input shapes is not of the form ``batch_shape_{0,1} +``
+            :attr:`input_shape`.
+        ValueError
+            If the inputs can not be broadcast to a common shape.
         """
 
-        if x1 is None:
-            try:
-                # Note that this differs from
-                # `np.broadcast_shapes(x0.shape, self._input_shape)`
-                # if self._input_shape contains `1`s
-                broadcast_input_shape = np.broadcast_to(
-                    x0,
-                    shape=x0.shape[: x0.ndim - self._input_ndim] + self._input_shape,
-                ).shape
-            except ValueError as ve:
-                err_msg = (
-                    f"The input array `x0` with shape {x0.shape} can not be broadcast "
-                    f"to the kernel's input shape {self.input_shape}."
-                )
-                raise ValueError(err_msg) from ve
-        else:
-            try:
-                np.broadcast_to(
-                    x0,
-                    shape=x0.shape[: x0.ndim - self._input_ndim] + self._input_shape,
-                )
+        err_msg = (
+            "The shape of the input array `{argname}` must match the `input_shape` "
```

Comment:
Are there some f-strings missing? The expressions `{argname}` and `{shape}` are not actually replaced by the corresponding variables.

Reply:
These are replaced using str.format below